### DISCUSSION TITLE
Should I reattach Folder event after reconnect?

### DISCUSSION THREAD

**Discussion by lazyclaw-labs** [jstedfast/MailKit#1698] [Q&A]
I have this event attahced to inbox:
	client.Inbox.CountChanged += (x, y) => hasnewmessage = true;

My questions, if I detect the client got disconnected, and I reconnect it immediately.
Now, do I need to reattach the event? I mean, will the inbox folder got reinitialized?

Thanks.

**Comment by jstedfast** [comment]
You do not need to reconnect any event handlers after reconnecting to the same server.

If you reconnect to a different server, the ImapClient's internal folder cache is cleared and new ImapFolder instances will be created.
